libminifi/include/core/ContentSession.h

```diff
@@ -39,6 +52,12 @@ class ContentSession {
   virtual void rollback() = 0;
 
   virtual ~ContentSession() = default;
+
+ protected:
+  virtual std::shared_ptr<io::BaseStream> append(const std::shared_ptr<ResourceClaim>& resource_id) = 0;
+
+  std::map<std::shared_ptr<ResourceClaim>, ExtensionData> extensions_;
```

Comment:
What is an extension, and why is it needed now?

Reply:
managing the append lock and the belonging append stream is now the responsibility of the ContentSession

Reply:
Why is it called extensions_? What is being extended?

Reply:
renamed to append_state_